src/runtime2/custom-element.js

```diff
@@ -380,7 +387,42 @@ function createBaseCustomElementClass(win) {
       this.implementation_.layout_ = this.layout_;
       this.implementation_.firstAttachedCallback();
       this.dispatchCustomEventForTesting(AmpEvents.ATTACHED);
-      this.getResources().upgraded(this);
+
+      // Block rendering of this component while it's DOM is parsing. This
+      // is technically not necessary for Bento components, but still needed
+      // for older components. And even for Bento components this has some
+      // performance benefits.
+      blockRender(this);
+
+      // By default, only build when the element can be rendered at all.
+      subscribe(this, [CanRender], (canRender) => {
+        if (canRender) {
```

Comment:
Yes. It's handled inside the `build()`. But this will be definitely getting more sophisticated. E.g. we may need the element's implementation to be refreshable. In which case it'd be `subscribe([CanRender, ImplClass])` or such. This part is still a bit murky.